docs/scos/dev/upgradability-services/upgradability-guidelines/entity-name-is-not-unique.md

```diff
@@ -0,0 +1,239 @@
+---
+title: Entity name is not unique
+description: Reference information for evaluator and upgrader tools.
+last_updated: Mar 23, 2022
+template: concept-topic-template
+---
+
+Modules have public and private APIs. While public API updates always support backward compatibility, private API updates can break backward compatibility. So, backward compatibility is not guaranteed in the private API. For example, if you use a core method on the project level, and it is updated or removed with an update, it can cause unexpected issues.
+
+For more information about module APIs, see [Definition of Module API](/docs/scos/dev/architecture/module-api/definition-of-module-api.html).
+
+When extending public API on the project level, make sure that entity names are unique, so Spryker updates are compatible with project changes. If a Spryker update introduces a core entity with a name matching a project-level entity name, their behavior might change or cause issues. To make your code unique, you can use prefixes like `Pyz` or your project name.
+
+The names of the following entities must be unique on the project level:
+
+* Transfers
+* Transfer properties
+* Database tables
+* Database columns
+* Methods
+* Constants
+
+## Transfer name is not unique
+
+Transfer names must be unique.
+
+### Example of code that causes an upgradability error: Transfer name is not unique
+
+```xml
+...
+    <transfer name="CustomProductData">
```

Comment:
Please add a real-life example here and in the rest of the document.

Reply:
Discussed. 
Will skip